q2_fmt/tests/test_engraftment.py

```diff
@@ -1250,10 +1254,336 @@ def test_rename_features_with_blank_label(self):
+                                              "donor2"]}
+                                     ).set_index('id')
+
+        donor_df = pd.DataFrame({
+            'id': ['donor2'],
+            'Feature1': [1],
+            'Feature2': [0],
+            'Feature3': [0]}).set_index('id')
+
+        exp_mask = [[1, 0, 0],
+                    [1, 0, 0],
+                    [1, 0, 0]]
+
+        donor_mask = _create_sim_masking(mismatched_df, donor_df,
+                                         reference_column='Ref')
+        np.testing.assert_array_equal(donor_mask, exp_mask)
+
+    def test_create_duplicated_table(self):
+        recip_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3'],
+            'Feature1': [1, 0, 0],
+            'Feature2': [0, 1, 0],
+            'Feature3': [0, 0, 1]}).set_index('id')
+
+        mismatched_df = pd.DataFrame({'id': ["sample1", "sample1",
+                                             "sample2", "sample2",
+                                             "sample3", "sample3"],
+                                      "Ref": ["donor2", "donor3",
+                                              "donor1", "donor3",
+                                              "donor1", "donor2"]}
+                                     ).set_index('id')
+
+        duplicated_recip_table = _create_duplicated_recip_table(mismatched_df,
+                                                                recip_df)
+
+        exp_d_r_table = pd.DataFrame({
+            'id': ['sample1', 'sample1', 'sample2', 'sample2',
+                   'sample3', 'sample3'],
+            'Feature1': [1, 1, 0, 0, 0, 0],
+            'Feature2': [0, 0, 1, 1, 0, 0],
+            'Feature3': [0, 0, 0, 0, 1, 1]}).set_index('id')
+
+        pd.testing.assert_frame_equal(duplicated_recip_table,
+                                      exp_d_r_table)
+
+    def test_create_no_duplicated_table(self):
+        recip_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3'],
+            'Feature1': [1, 0, 0],
+            'Feature2': [0, 1, 0],
+            'Feature3': [0, 0, 1]}).set_index('id')
+
+        mismatched_df = pd.DataFrame({'id': ["sample1",
+                                             "sample2",
+                                             "sample3"],
+                                      "Ref": ["donor2",
+                                              "donor2",
+                                              "donor2"]}
+                                     ).set_index('id')
+
+        duplicated_recip_table = _create_duplicated_recip_table(mismatched_df,
+                                                                recip_df)
+
+        exp_d_r_table = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3'],
+            'Feature1': [1, 0, 0],
+            'Feature2': [0, 1, 0],
+            'Feature3': [0, 0, 1]}).set_index('id')
+
+        pd.testing.assert_frame_equal(duplicated_recip_table,
```

Comment:
Your expected value here could just be `recip_df`, in which case you wouldn't have to define `exp_d_r_table`.